packages/calcite-components/src/components/tab-nav/tab-nav.e2e.ts

```diff
@@ -192,4 +174,96 @@ describe("calcite-tab-nav", () => {
     await page.keyboard.press("Home");
     expect(await page.evaluate(() => document.activeElement.id)).toBe("tab1");
   });
+
+  const inlineTabsWithVariedTitleLength = html`
+    <calcite-tabs layout="inline" style="margin: 75px; width: 375px">
+      <calcite-tab-nav slot="title-group" id="testSubjectNav">
+        <calcite-tab-title selected icon-start="tabbed-view" icon-end="pen" closable> Tab 1 Title </calcite-tab-title>
+        <calcite-tab-title icon-start="tabbed-view">Tab 2 Title</calcite-tab-title>
+        <calcite-tab-title>An Ultramarathon of a Tab Title, why not.</calcite-tab-title>
+        <calcite-tab-title closable>Tab 4 Title</calcite-tab-title>
+        <calcite-tab-title>Tab 5 Title</calcite-tab-title>
+        <calcite-tab-title>Tab 6 Title</calcite-tab-title>
+      </calcite-tab-nav>
+      <calcite-tab selected>Tab 1 Content</calcite-tab>
+      <calcite-tab>Tab 2 Content</calcite-tab>
+      <calcite-tab>Tab 3 Content</calcite-tab>
+      <calcite-tab>Tab 4 Content</calcite-tab>
+      <calcite-tab>Tab 5 Content</calcite-tab>
+      <calcite-tab>Tab 6 Content</calcite-tab>
+    </calcite-tabs>
+  `;
+
+  describe("responsive tabs for inline layout", () => {
+    const overflowScenarios = ["end", "start", "both"];
+    let page: E2EPage;
+
+    beforeEach(async () => {
+      page = await newE2EPage();
+      await page.setContent(inlineTabsWithVariedTitleLength);
+    });
+
+    it("should overflow tab-titles that don't fit within the bounds of tab-nav", async () => {
+      const { tabTitlesTotalWidth, tabNavWidth } = await page.evaluate(() => {
+        const tabNav = document.getElementById("testSubjectNav") as HTMLCalciteTabNavElement;
+        const tabTitles = Array.from(document.querySelectorAll("calcite-tab-title")) as HTMLCalciteTabTitleElement[];
+
+        const tabNavWidth = tabNav.offsetWidth;
+        const tabTitlesTotalWidth = tabTitles.reduce((sum, tabTitle: HTMLElement) => {
+          return sum + tabTitle.offsetWidth;
+        }, 0);
+
+        return { tabTitlesTotalWidth, tabNavWidth };
+      });
+
+      expect(tabTitlesTotalWidth).toBeGreaterThan(tabNavWidth);
+    });
+
+    overflowScenarios.forEach(async (overflowScenario) => {
+      if (overflowScenario === "end") {
```

Comment:
Would it be possible to add the specific scenarios the test needs to cover vs conditionally running different tests per scenario? Something similar to [this (note that it has some pending cleanup 😅)](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/tree/tree.e2e.ts#L1148). This would avoid disabling the `jest/no-conditional-expect` rule.